src/diff/mount.js

```diff
@@ -188,6 +188,15 @@ function mountDOMElement(dom, internal, globalContext, commitQueue) {
 		}
 
 		let newHtml, newValue, newChildren;
+		if (internal.type === 'input') {
```

Comment:
When we create the new component we see whether it has a `checked` or `value` property, this will make it controlled